mantidproject/mantid
# Add run numbers to NXcanSAS and canSAS metadata

**Description of work.**
This PR adds "additional" run numbers to NXcanSAS and CanSAS files. Can transmission and sample scatter are already added to these files, so this PR adds sample direct, sample transmission, can direct, and can scatter.

**Report to:** Steve King ISIS

**To test:**
1. Interfaces -> SANS -> ISIS SANS
2. Load the attached user and batch file
3. Select an appropriate output directory in Directories Manager
4. In the settings tab (on the left), change reduction mode (at the top) to **front**
5. Back in the runs tab, select file output mode radio button (at the bottom), and select the NXCanSAS and CanSAS output file types
6. Click process all
7. Using your favourite hdf viewing software, open **aa.h5**
8. Navigate to sasEntry/sasProcess/sasNote. The datasets should match up the run numbers in the sans table, excluding sample scatter and can transmission
9. Open up aa.xml and scroll to the bottom. There should be terms containing the run numbers in the sans table, exluding sample scatter and can transmission

Fixes #22813 

**Data:**
[metadata_pr.zip](https://github.com/mantidproject/mantid/files/3325503/metadata_pr.zip)

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
